### PR TITLE
fetch Allo V2 programs from the indexer

### PR DESCRIPTION
**Pull request by bhargavaparoksham**
Fixes: #2816 

## Description

fetch Allo V2 programs from the indexer

## Checklist

This PR:

- [ ] Includes only changes relevant to the original ticket. Significant refactoring needs to be separated.
- [ ] Doesn't disable eslint rules.
- [ ] Doesn't work around the type checker (including but not limited to: type casts, non-null assertions, `@ts-ignore`, unjustified optional values).
- [ ] Doesn't contain commented out code.
- [ ] If adding/updating a feature, it adds/updates its test script on Notion.

### CODE REVIEW

File: packages/data-layer/src/queries.ts [hunk 1]
```diff
@@ -31,6 +30,37 @@ export const getProgramsByUser = gql`
...
 
+/**
+ * Get all the profiles that a user is a part of in allo v2
+ * @param $alloVersion - The version of Allo
+ * @param $address - The address of the user
+ * @param $chainId - The network ID of the chain
+ *
+ * @returns The programs
+ */
+export const getProfilesByUser = gql`
```
Comment (codenamejason) [inline packages/data-layer/src/queries.ts:41]: We have one for projects below called getProjectsByAddress that is very similar - I think our naming is starting to bleed about.

**Review by vacekj — APPROVED** [review]

**Review by michellema1208 — COMMENTED** [review]
hey @bhargavaparoksham - I'm able to see the profile I created on Builder on Manager, but not able to open the profile on Manager. Is that expected / out of scope for this PR? Otherwise, I would expect to be able to open the profile and be able to create rounds under it (once the "create round on v2" PR is ready). 

Screen Shot 2024-02-12 at 10 59 00 AM

Screen Shot 2024-02-12 at 10 58 48 AM

Manager link: https://manager-staging-git-manager-2816-grants-stack.vercel.app/#/program/0x16aaa61bcdf4445bad79cc671fa513c357b218b13720d6e2adb70af404d2007b

File: packages/data-layer/src/queries.ts [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
- * Get all the programs that a user is a part of
- * @param $alloVersion - The version of Allo
+ * Manager: Get all the programs that a user is a part of in allo v1
  * @param $address - The address of the user
  * @param $chainId - The network ID of the chain
  *
  * @returns The programs
  */
-export const getProgramsByUser = gql`
+export const getV1ProjectsByUser = gql`
```
Comment (boudra) [inline packages/data-layer/src/queries.ts:10]: these two separate queries are exactly the same you could have a single query here that receives a tag to filter by, the caller can use program for v1 or allo-v2 for v2

**Review by boudra — CHANGES_REQUESTED** [review]
not sure i understand why both queries that are exactly the same, have different types and if statements, where one is being mapped to the exact same type, it feels like there's a lot of repetition that could cause human error if someone would forget to update a type or a query

would something like this work?

```typescript
const getProgramsByUserAndTag = gql`
  query ($address: String!, $chainId: Int!, $filterByTag: String ) {
    projects(
      filter: {
        tags: { contains: $filterByTag }
        roles: { some: { address: { equalTo: $address } } }
        chainId: { equalTo: $chainId }
      }
    ) {
      id
      chainId
      metadata
      metadataCid
      tags
      roles {
        address
        role
        createdAtBlock
      }
    }
  }
`;

let response: { projects: Program[] } = { projects: [] };

response = await request(
  this.gsIndexerEndpoint,
  getProgramsByUserAndTag,
  { ...requestVariables, filterByTag: alloVersion === "allo-v2" ? "allo-v2" : "program" },
);

return response.projects;
```

**Review by boudra — APPROVED** [review]

**Pull request merged**